### DISCUSSION TITLE
wrong jodit-workplace height when toolbar is set to an HTMLElement

### DISCUSSION THREAD

**Discussion by intellysoft** [xdan/jodit#920] [Q&A] [answered]
as indicated in the title, when the toolbar is set to an element, the height dimension of the workplace indicated in the style is wrong - in essence it is reduced by the height of the toolbar anyway... how could I get around this, without changing the source code ?

**Comment by xdan** [comment]
When the toolbar is mounted into a separate element, the `height` you set still applies to the editor container, and the workplace ends up shrunk by the toolbar height that isn't even inside it. The quickest workaround without touching the source is to set the height on the workplace directly in CSS — `.jodit-workplace { height: 400px !important }` (or whatever you need) — instead of via the `height` option.

If that doesn't hold for your layout, tell me how you're mounting the toolbar and I'll take a closer look — it may be worth a fix.

**Comment by xdan** [comment, marked as answer]
Fixed properly in 4.12.32. The size logic was subtracting the toolbar's height from the workplace even when the toolbar is rendered into an external element — so it shrank the area by a toolbar that wasn't actually inside the container. Now the toolbar height is only subtracted when the toolbar really lives inside the container, so the height you set is respected. Upgrade and you can drop the CSS workaround.
